docs/source/concepts/interoperability.rst

```diff
@@ -0,0 +1,237 @@
+Accordingly, the ledger-aware projections look as follows, where yellow represents the `Iou` ledger and blue stands for the painting ledger.
+Here, the projections to the blue ledger include the actions of the yellow ledger because a projection includes the subactions.
+
+.. https://www.lucidchart.com/documents/edit/f8ec5741-7a37-4cf5-92a9-bf7b3132ba8e
+.. image:: ./images/projecting-transactions-paint-offer-ledger-aware.svg
+   :align: center
+   :width: 60%
+
+Definition »Y-labelled action«
+  An action is **Y-labelled** for a set `Y` if it is labelled with an element of `Y`.
+           
+Definition »ledger-aware projection«
+  Let `tx` be a transaction according to the DAML Ledger Model whose actions are `X`\ -labelled for a set `X` of DAML Ledgers
+  and let `Y` be a subset of `X`.
+  Let `Act` be the set of `Y`\ -labelled subactions of `tx` that have `A` as an informee.
+  The **Y-projection** of `tx` to a party `A` consists of maximal elements of `Act` (w.r.t. the subaction relation) in execution order.
+
+Ledger-aware projections extend to the virtual shared ledger straightforwardly.
+  
+Interoperable order consistency
+===============================
+  
+The interoperable order consistency between local ledgers is then defined as follows:
+  
+Definition »interoperable order consistency«
+  A set `Ls` of local ledgers is **interoperable order-consistent**
+  if there exists a virtual shared ledger `G` for a set of `X` of interoperable DAML Ledgers with the following properties:
+
+  * *Closed world:*
+    The ledgers in `Ls` aggregate only DAML Ledgers in `X`.
+
+  * *Injective agreement:*
+    For every local ledgers `L` in `Ls` for party `A` that aggregates a set `Y` of DAML ledgers, the following holds.
+    Let `V` be the vertices in `L` that contain actions other than **Enter** and **Leave**
+    and let `W` be the vertices in `G` that whose `Y`\ -projection to `A` is not empty.
+    Then there is a bijection between `V` and `W` such that removing all **Enter** and **Leave** actions from the transaction in `V` gives the `Y`-projection of the corresponding transaction in `W`.
+    Whenver `L` contains an edge between two in `V`, then `G` contains an edge between the corresponding vertices in `W`.
+
+  * *Enter and Leave*:
+    For every `Y`\ -labelled action `act` in `G` with informee `A` that uses a contract `c`, the following holds:
+
+    - If there is no `Y`\ -labelled action that precedes `act` in `G`\ 's action order and acts on the same contract,
+      then `L` contains an **Enter** action for the contract that precedes `act` in `L`.
+
+    - If there are actions `act`:sub:`1` and `act`:sub:`2` such that `act`:sub:`1` precedes `act`:sub:`2` and `act`:sub:`2` precedes `act` in `G` and `act`:sub:`1` is `Y`\ -labelled, but not `act`:sub:`2`, then `L` contains an **Enter** and a **Leave** action for the contract between `act`:sub:`1` and `act`.
```

Comment:
This is pretty complex. Since the virtual shared ledger already contains ledger labels, I'd find it much simpler to just add the `Enter L` and `Leave L` actions to it. You also probably want some kind of consistency criterion on that, i.e., actions on a contract can't just jump around ledgers willy-nilly.